catalog/spatial/csw/spatial-csw-transformer/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/converter/XstreamPathConverter.java

```diff
@@ -164,13 +166,11 @@ protected boolean doBasicPathsMatch(final Path pathObj1, final Path pathObj2) {
      */
     private boolean endsMatch(char[] path1, int index1, char[] path2, int index2) {
```

Comment:
That was my original approach. It worked pretty well (ran at about 1/3rd the original runtime), but a single pass approach reduced that improved runtime by another 1/3rd down to < 1/5th the original runtime. I think that kind of increase is worth it.
I agree this is a considerably more cognitively complex, but this change isolates the tag skipping logic from the attribute handling (where I had foolishly combined them before -resulting in the need for this fix).
As far as caching goes, I think that would be worth looking into.